scripts/markdown_seo_lint.js

```diff
@@ -0,0 +1,219 @@
+const fs = require("fs");
+const yaml = require("js-yaml");
+const markdownlint = require("markdownlint");
+const path = require("path");
+
+const FRONT_MATTER_REGEX = /((^---\s*$[^]*?^---\s*$)|(^\+\+\+\s*$[^]*?^(\+\+\+|\.\.\.)\s*$))(\r\n|\r|\n|$)/m;
+
+function checkPageTitle(title) {
+    let error = null;
+    if (!title) {
+        error = "Missing page title";
+    }else if (title.trim().length === 0) {
```

Comment:
Nit: Rather than call `title.trim().length` twice, we can just rewrite this as the following. Also, IMHO it seems a little cleaner to just return the string literal rather than assign it to `error` and return that.

```
let titleLength = title.trim().length;
if (titleLength === 0) {
    return "Page title is empty";
}
if (titleLength > 60) {
    return "Page title exceeds 60 characters";
}
return null;
```

Reply:
Sounds good. Though to write it like this we will first need to check the title is not `null` or `undefined` otherwise the script will throw an error.

```
    if (!title) {
        return "Missing page title";
    }else if (typeof title === "string") {
        const titleLength = title.trim().length;
        if (titleLength === 0) {
            return "Page title is empty";
        }else if (titleLength > 60) {
            return "Page title exceeds 60 characters";
        }
    }
    return null;
```